capabilities/capabilities.go

```diff
@@ -0,0 +1,105 @@
+// Package capabilities provides an API for capability detection.
+package capabilities
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/giantswarm/microerror"
+)
+
+var (
+	// Autoscaling is the capability to scale tenant clusters automatically.
+	Autoscaling = CapabilityDefinition{
+		Name: "Autoscaling",
+		RequiredReleasePerProvider: []ReleaseProviderPair{
+			ReleaseProviderPair{
+				Provider:       "aws",
+				ReleaseVersion: semver.MustParse("6.3"),
+			},
+		},
+	}
+
+	// AvailabilityZones is the capability to spread the worker nodes of a tenant
+	// cluster over multiple availability zones.
+	AvailabilityZones = CapabilityDefinition{
+		Name: "AvailabilityZones",
+		RequiredReleasePerProvider: []ReleaseProviderPair{
+			ReleaseProviderPair{
+				Provider:       "aws",
+				ReleaseVersion: semver.MustParse("6.1"),
+			},
+		},
+	}
+
+	// NodePools is the capabilitiy to group tenant cluster workers logically.
+	NodePools = CapabilityDefinition{
+		Name: "NodePools",
+		RequiredReleasePerProvider: []ReleaseProviderPair{
+			ReleaseProviderPair{
+				Provider:       "aws",
+				ReleaseVersion: semver.MustParse("9"), // TODO: fix once the node pools release version is defined.
+			},
+		},
+	}
+
+	// AllCapabilityDefinitions contains all the capabilities
+	AllCapabilityDefinitions = []CapabilityDefinition{
+		Autoscaling,
+		AvailabilityZones,
+		NodePools,
+	}
+)
+
+// CapabilityDefinition is the type we use to describe the conditions that have to be met
+// so that we can assume a certain capability on the installation or API side.
+type CapabilityDefinition struct {
+	Name        string
+	Description string
+	// RequiredReleasePerProvider holds the combination(s) of provider and
+	// release version which have to be fulfilled so we assume a capability.
+	RequiredReleasePerProvider []ReleaseProviderPair
+}
+
+// ReleaseProviderPair is a combination of a providr ('aws', 'azure', 'kvm) and
+// a release version number.
+type ReleaseProviderPair struct {
+	Provider       string
+	ReleaseVersion *semver.Version
+}
+
+// GetCapabilities returns the capabilities available in the current context
+func GetCapabilities(provider, releaseVersion string) ([]CapabilityDefinition, error) {
+	cap := []CapabilityDefinition{}
```

Comment:
```suggestion
	capability := []CapabilityDefinition{}
```

`cap` is a reserved word for checking the capacity of a slice

Reply:
Changed